sled-agent/src/bootstrap/server.rs

```diff
@@ -276,110 +179,19 @@ impl Inner {
-            let trust_quorum_share =
-                trust_quorum_share.map(ShareDistribution::from);
-
-            // Save the trust quorum share _before_ calling request_agent, so
-            // that we can return it in the `Request::ShareRequest` branch below
-            //
-            // TODO should we check that if `initial_share` is already
-            // `Some(_)`, the value is the same? If it's not, we got two
-            // different shares from RSS...
-            *initial_share.lock().await = trust_quorum_share.clone();
-
-            match bootstrap_agent
-                .request_agent(&request, &trust_quorum_share)
-                .await
-            {
-                Ok(response) => {
-                    // If this send fails, it means our caller already received
-                    // our share from a different
-                    // `serve_request_before_quorum_initialization()` task
-                    // (i.e., from another incoming request from RSS). We'll
-                    // ignore such failures.
-                    let _ = tx_share.send(trust_quorum_share).await;
-
-                    Ok(Response::SledAgentResponse(response))
-                }
-                Err(err) => {
-                    warn!(log, "Sled agent request failed"; "err" => %err);
-                    Err(format!("Sled agent request failed: {err}"))
-                }
-            }
-        }
-        Request::ShareRequest => match initial_share.lock().await.clone() {
-            Some(dist) => Ok(Response::ShareResponse(dist.share)),
-            None => {
-                warn!(log, "Share requested before we have one");
-                Err("Share request failed: share unavailable".to_string())
-            }
-        },
-    };
-
-    write_response(&mut stream, response).await
-}
-
 async fn serve_request_after_quorum_initialization(
     stream: TcpStream,
     bootstrap_agent: Arc<Agent>,
-    // TODO-cleanup `sp` and `tx_share` are optional while we still allow
-    // trust-quorum-free dev/test setups. Eventually they should be required.
-    sp: Option<SpHandle>,
-    trust_quorum_share: &Option<ShareDistribution>,
     log: &Logger,
 ) -> Result<(), String> {
-    // Establish sprockets session (if we have an SP).
-    let mut stream = crate::sp::maybe_wrap_stream(
-        stream,
-        &sp,
-        SprocketsRole::Server,
-        trust_quorum_share
-            .as_ref()
-            .map(|dist| dist.member_device_id_certs.as_slice()),
-        log,
-    )
-    .await
-    .map_err(|err| format!("Failed to establish sprockets session: {err}"))?;
+    let mut stream = Box::new(tokio::io::BufStream::new(stream));
 
     let response = match read_request(&mut stream).await? {
-        Request::SledAgentRequest(request, trust_quorum_share) => {
-            let trust_quorum_share =
-                trust_quorum_share.map(ShareDistribution::from);
-
+        Request::SledAgentRequest(request) => {
```

Comment:
We should probably also change the name of this request to `StartSledAgentRequest`